portal-kernel/src/com/liferay/portal/kernel/internal/service/persistence/TableMapperImpl.java

```diff
@@ -444,12 +447,23 @@ protected static <T extends BaseModel<T>> List<T> getBaseModels(
 			return Collections.emptyList();
 		}
 
+		long currentCompanyId = CompanyThreadLocal.getCompanyId();
 		List<T> slaveBaseModels = new ArrayList<>(slavePrimaryKeys.length);
 
 		try {
 			for (long slavePrimaryKey : slavePrimaryKeys) {
-				slaveBaseModels.add(
-					slaveBasePersistence.findByPrimaryKey(slavePrimaryKey));
+				T slaveBaseModel =
+					slaveBasePersistence.findByPrimaryKey(slavePrimaryKey);
+
+				long checkableCompanyId =
+					_getCheckableCompanyId(slaveBaseModel);
```

Comment:
@gianmarcobrunialti, to improve the performance, can you add the check __getCheckableCompanyId_ inline in the if so we only execute it when needed?